src/trim_comparison.rs

```diff
@@ -0,0 +1,808 @@
+                "native": "0x16522c650ca4f"
+              }
+            },
+          ],
+        });
+
+        let config = TrimConfig { level: 2 };
+        let trimmed_obj = trim_comparison(obj, &config);
+        assert_eq!(trimmed_obj, expected_trimmed_obj);
+    }
+
+    #[test]
+    fn test_same() {
+        let obj = json!({
+          "first_element": "Same(0x1176a1bd84444c89232ec27754698e5d2e7e1a7f1539f12027f28b23ec9f3d8)",
+          "second_element":  {
+              "Different": {
+                "base": "0x2863eece73692",
+                "native": "0x16522c650ca4f"
+              }
+            },
+        });
+
+        let expected_trimmed_obj = json!({
+          "second_element":  {
+              "Different": {
+                "base": "0x2863eece73692",
+                "native": "0x16522c650ca4f"
+              }
+            },
+        });
+
+        let config = TrimConfig { level: 0 };
+        let trimmed_obj = trim_comparison(obj, &config);
+        assert_eq!(trimmed_obj, expected_trimmed_obj);
+    }
+
+    #[test]
+    fn test_simple_2() {
+        let obj = json!({
+          "block_num": 647485,
+          "calldata": [
+            "Same(0x1176a1bd84444c89232ec27754698e5d2e7e1a7f1539f12027f28b23ec9f3d8)",
+            {
+              "Different": {
+                "base": "0x2863eece73692",
+                "native": "0x16522c650ca4f"
+              }
+            },
+            "Same(0x0)"
+          ],
+        });
+
+        let expected_trimmed_obj = json!({
+          "block_num": 647485,
+          "calldata": [
+            "Same(0x1176a1bd84444c89232ec27754698e5d2e7e1a7f1539f12027f28b23ec9f3d8)",
+            {
+              "Different": {
+                "base": "0x2863eece73692",
+                "native": "0x16522c650ca4f"
+              }
+            },
+            "Same(0x0)"
+          ],
+        });
+
+        let config = TrimConfig { level: 2 };
+        let trimmed_obj = trim_comparison(obj, &config);
+        assert_eq!(trimmed_obj, expected_trimmed_obj);
+    }
+
+    #[test]
+    fn test_simple_0() {
+        let obj = json!({
+          "block_num": 647485,
+          "calldata": [
+            "Same(0x1176a1bd84444c89232ec27754698e5d2e7e1a7f1539f12027f28b23ec9f3d8)",
+            {
+              "Different": {
+                "base": "0x2863eece73692",
+                "native": "0x16522c650ca4f"
+              }
+            },
+            "Same(0x0)"
+          ],
+        });
+
+        let expected_trimmed_obj = json!({
+          "block_num": 647485,
+          "calldata": [
+            {
+              "Different": {
+                "base": "0x2863eece73692",
+                "native": "0x16522c650ca4f"
+              }
+            },
+          ],
+        });
+
+        let config = TrimConfig { level: 0 };
+        let trimmed_obj = trim_comparison(obj, &config);
+        assert_eq!(trimmed_obj, expected_trimmed_obj);
+    }
+
+    #[test]
+    fn test_many() {
```

Comment:
Reduce the size of this test a bit

Reply:
Removed test_many since it's covered by `test_long_array` already.